.pre-commit-config.yaml

```diff
@@ -27,3 +27,14 @@ repos:
       - id: flake8
         additional_dependencies: [pydocstyle>5.1.0, flake8-docstrings>1.4.0]
         args: ["--docstring-convention=all"]
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.1.0
+    hooks:
+      - id: codespell
+        files: ^.*\.(py|c|h|md|rst|yml)$
+        args: [
+          "--ignore-words",
+          "ci/codespell-ignore-words.txt",
```

Comment:
The list hopefully doesn't seem too excessive, but as Matplotlib has a lot of abbreviations of variables in it this is the minimum number of ignores needed to get `codespell` working when run across all of the above file types.